mantidproject/mantid
# Remove IPropertyManager::filterByTime method

Since `IPropertyManager::filterByTime()` is only actually implemented by `LogManager`, remove it from the base class. The other part is changing the implementation to creating/setting the `TimeROI` object rather than modifying all of the logs and adding artificial values to them. The previous behavior was to filter things [inclusive, inclusive] for the logs and [inclusive, exclusive) for events. The `TimeROI` changes the behavior for the logs so they are consistent with the event filtering. Only proton charge appears to have occurrences where this behavior change has effect. A handful of system tests that filter which were updated. This also fixes #35361, by finally correcting the calculation of the integrated proton charge.

**To test:**

This is a refactor so reviewing the code is probably the better way to verify the changes.

Refs #34794.

*This does not require release notes* because a larger set of release notes will be added in a future PR before the release is created.



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
